### PR TITLE
Two internal/API features for Lists

### PR DESCRIPTION
**Pull request by srawlins**
Two internal/API features for Lists:

* Add separator character to ListSeparator, similar to BinaryOperator.
* Add trailing comma to ListExpression.

### CODE REVIEW

**Comment by nex3** [comment]
Sorry for the long wait time for a review! I've been on medical leave for the past couple months.

I don't understand the trailing comma change here. Does it just affect the `inspect()` output? I don't think tracking whether the user wrote a trailing comma is consistent with Ruby Sass.

**Comment by srawlins** [comment]
Ah sorry, the trailing comma change is just so that ListExpression exposes the notion externally, not for use in compiling to CSS.

I can table that part of this change for now. It's part of a bigger effort I'd like to engage in, where I'd like the AST to be a more compete representation of the source. We can chat offline.

**Comment by nex3** [comment]
> I can table that part of this change for now. It's part of a bigger effort I'd like to engage in, where I'd like the AST to be a more compete representation of the source. We can chat offline.

Ah, I see. Yeah, let's VC about that.

**Comment by srawlins** [comment]
OK reduced changes to just adding the separator character.

File: lib/src/value/list.dart [hunk 1]
```diff
@@ -57,20 +57,22 @@ class SassList extends Value {
...
   ///
   /// Singleton lists and empty lists don't have separators defiend. This means
   /// that list functions will prefer other lists' separators if possible.
-  static const undecided = const ListSeparator._("undecided");
+  static const undecided = const ListSeparator._("undecided", null);
 
   final String _name;
 
-  const ListSeparator._(this._name);
+  final String separator;
```
Comment (nex3) [inline lib/src/value/list.dart]: Document this. In particular, mention that it's `null` sometimes since that's not necessarily obvious from the API.

Comment (srawlins) [inline lib/src/value/list.dart]: Done.

**Comment by nex3** [comment]
Thanks!

**Pull request merged**